manager/assets/modext/widgets/core/tree/modx.tree.js

```diff
@@ -145,13 +145,50 @@ Ext.extend(MODx.tree.Tree,Ext.tree.TreePanel,{
     ,options: {}
     ,disableHref: false
 
-    ,onLoad: function(ldr,node,resp) {
+    ,onLoad: function (ldr,node,resp) {
         // no select() here, just addClass, using Active Input Cookie Value to set focus
-        Ext.each(node.childNodes, function(node){
+        Ext.each(node.childNodes,function (node) {
             if (node.attributes.selected) {
                 //node.select();
                 node.ui.addClass('x-tree-selected');
             }
+
+            var elId = node.ui.elNode.id + '_tools';
+            var el = document.createElement('div');
+            el.id = elId;
+            el.className = 'modx-tree-node-btn-create';
+
+            // for now we only add a create icon to documents
+            if (node.attributes.classKey == "modDocument") {
+                node.ui.elNode.appendChild(el);
+
+                var btn = MODx.load({
+                    xtype: 'modx-button',
+                    text: '',
+                    scope: this,
+                    tooltip: new Ext.ToolTip({
+                        // TODO use dictionary here
+                        // TODO if childtemplate property is available, directly use that instead of "document"
+                        title: _('create_document_inside')+" <strong>" + node.attributes.text + "</strong>"
+                        ,target: this
+                    }),
+                    node: node,
+                    handler: function (btn,evt) {
+                        evt.stopPropagation(evt);
+                        node.getOwnerTree().handleDirectCreateClick(node);
+                    },
+                    iconCls: 'icon-plus-square',
+                    renderTo: elId,
+                    listeners: {
```

Comment:
Have you tried, instead of the `tooltip` + `listeners`, to add a `description`? I think that'd normally handle the showing/hiding automatically, so might save some code..

Reply:
To be honest, that code was a copy of the context btn code. That should be changed there as well then. I will try that.

Reply:
Fair enough! ;)

Reply:
Using a description instead of tooltip+listeners does not show up anything. Any idea?